From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Davis Goodin <dagood@microsoft.com>
Date: Thu, 14 Oct 2021 18:24:18 -0500
Subject: [PATCH] Build more projects during source-build

Use Roslyn.sln, not Compilers.sln, to build more projects during source-build.
Update ExcludeFromSourceBuild properties to include more projects and exclude a
few projects that shouldn't be in source-build.
https://github.com/dotnet/roslyn/pull/57165

The newly included projects are used by downstream repos.
---
 eng/SourceBuild.props                                       | 2 +-
 .../Core/Tests}/Directory.Build.props                       | 0
 .../CodeAnalysis.Debugging}/Directory.Build.props           | 0
 src/Dependencies/Microsoft.NetFX20/Directory.Build.props    | 6 ++++++
 src/Dependencies/PooledObjects/Directory.Build.props        | 6 ++++++
 src/Features/LanguageServer/Directory.Build.props           | 6 ++++++
 src/Features/Lsif/Directory.Build.props                     | 6 ++++++
 src/NuGet/VisualStudio/Directory.Build.props                | 6 ++++++
 .../Microsoft.CodeAnalysis.Workspaces.MSBuild.csproj        | 2 --
 9 files changed, 31 insertions(+), 3 deletions(-)
 rename src/{Dependencies => CodeStyle/Core/Tests}/Directory.Build.props (100%)
 rename src/{Features => Dependencies/CodeAnalysis.Debugging}/Directory.Build.props (100%)
 create mode 100644 src/Dependencies/Microsoft.NetFX20/Directory.Build.props
 create mode 100644 src/Dependencies/PooledObjects/Directory.Build.props
 create mode 100644 src/Features/LanguageServer/Directory.Build.props
 create mode 100644 src/Features/Lsif/Directory.Build.props
 create mode 100644 src/NuGet/VisualStudio/Directory.Build.props

diff --git a/eng/SourceBuild.props b/eng/SourceBuild.props
index 182f8a9cdb1..92e316a4744 100644
--- a/eng/SourceBuild.props
+++ b/eng/SourceBuild.props
@@ -11,7 +11,7 @@
   -->
   <Target Name="ConfigureInnerBuildArg" BeforeTargets="GetSourceBuildCommandConfiguration">
     <PropertyGroup>
-      <InnerBuildArgs>$(InnerBuildArgs) /p:Projects="$(InnerSourceBuildRepoRoot)\Compilers.sln"</InnerBuildArgs>
+      <InnerBuildArgs>$(InnerBuildArgs) /p:Projects="$(InnerSourceBuildRepoRoot)\Roslyn.sln"</InnerBuildArgs>
     </PropertyGroup>
   </Target>
 
diff --git a/src/Dependencies/Directory.Build.props b/src/CodeStyle/Core/Tests/Directory.Build.props
similarity index 100%
rename from src/Dependencies/Directory.Build.props
rename to src/CodeStyle/Core/Tests/Directory.Build.props
diff --git a/src/Features/Directory.Build.props b/src/Dependencies/CodeAnalysis.Debugging/Directory.Build.props
similarity index 100%
rename from src/Features/Directory.Build.props
rename to src/Dependencies/CodeAnalysis.Debugging/Directory.Build.props
diff --git a/src/Dependencies/Microsoft.NetFX20/Directory.Build.props b/src/Dependencies/Microsoft.NetFX20/Directory.Build.props
new file mode 100644
index 00000000000..6eef643958f
--- /dev/null
+++ b/src/Dependencies/Microsoft.NetFX20/Directory.Build.props
@@ -0,0 +1,6 @@
+<Project>
+  <Import Project="$([MSBuild]::GetPathOfFileAbove('Directory.Build.props', '$(MSBuildThisFileDirectory)../'))" />
+  <PropertyGroup>
+    <ExcludeFromSourceBuild>true</ExcludeFromSourceBuild>
+  </PropertyGroup>
+</Project>
diff --git a/src/Dependencies/PooledObjects/Directory.Build.props b/src/Dependencies/PooledObjects/Directory.Build.props
new file mode 100644
index 00000000000..6eef643958f
--- /dev/null
+++ b/src/Dependencies/PooledObjects/Directory.Build.props
@@ -0,0 +1,6 @@
+<Project>
+  <Import Project="$([MSBuild]::GetPathOfFileAbove('Directory.Build.props', '$(MSBuildThisFileDirectory)../'))" />
+  <PropertyGroup>
+    <ExcludeFromSourceBuild>true</ExcludeFromSourceBuild>
+  </PropertyGroup>
+</Project>
diff --git a/src/Features/LanguageServer/Directory.Build.props b/src/Features/LanguageServer/Directory.Build.props
new file mode 100644
index 00000000000..6eef643958f
--- /dev/null
+++ b/src/Features/LanguageServer/Directory.Build.props
@@ -0,0 +1,6 @@
+<Project>
+  <Import Project="$([MSBuild]::GetPathOfFileAbove('Directory.Build.props', '$(MSBuildThisFileDirectory)../'))" />
+  <PropertyGroup>
+    <ExcludeFromSourceBuild>true</ExcludeFromSourceBuild>
+  </PropertyGroup>
+</Project>
diff --git a/src/Features/Lsif/Directory.Build.props b/src/Features/Lsif/Directory.Build.props
new file mode 100644
index 00000000000..6eef643958f
--- /dev/null
+++ b/src/Features/Lsif/Directory.Build.props
@@ -0,0 +1,6 @@
+<Project>
+  <Import Project="$([MSBuild]::GetPathOfFileAbove('Directory.Build.props', '$(MSBuildThisFileDirectory)../'))" />
+  <PropertyGroup>
+    <ExcludeFromSourceBuild>true</ExcludeFromSourceBuild>
+  </PropertyGroup>
+</Project>
diff --git a/src/NuGet/VisualStudio/Directory.Build.props b/src/NuGet/VisualStudio/Directory.Build.props
new file mode 100644
index 00000000000..6eef643958f
--- /dev/null
+++ b/src/NuGet/VisualStudio/Directory.Build.props
@@ -0,0 +1,6 @@
+<Project>
+  <Import Project="$([MSBuild]::GetPathOfFileAbove('Directory.Build.props', '$(MSBuildThisFileDirectory)../'))" />
+  <PropertyGroup>
+    <ExcludeFromSourceBuild>true</ExcludeFromSourceBuild>
+  </PropertyGroup>
+</Project>
diff --git a/src/Workspaces/Core/MSBuild/Microsoft.CodeAnalysis.Workspaces.MSBuild.csproj b/src/Workspaces/Core/MSBuild/Microsoft.CodeAnalysis.Workspaces.MSBuild.csproj
index 3bd6d6398e3..2edf2da6d1a 100644
--- a/src/Workspaces/Core/MSBuild/Microsoft.CodeAnalysis.Workspaces.MSBuild.csproj
+++ b/src/Workspaces/Core/MSBuild/Microsoft.CodeAnalysis.Workspaces.MSBuild.csproj
@@ -7,8 +7,6 @@
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
     <TargetFrameworks>netcoreapp3.1;net472</TargetFrameworks>
     <DefineConstants>$(DefineConstants);WORKSPACE_MSBUILD</DefineConstants>
-    <ExcludeFromSourceBuild>true</ExcludeFromSourceBuild>
-
     <!-- NuGet -->
     <IsPackable>true</IsPackable>
     <PackageDescription>
